backend/compact-connect/lambdas/staff-users/data_model/client.py

```diff
@@ -109,61 +109,116 @@ def update_user_permissions(
         logger.info('Updating staff user permissions', user_id=user_id)
 
         update_expression_parts = {'add': [], 'delete': []}
+
+        # DynamoDB does not support both ADD and DELETE on the same String Set in a single UpdateItem call, so we will
+        # split additions and removals into two calls to prevent a conflict.
+        resp = self._handle_user_permission_additions(
+            user_id=user_id,
+            compact=compact,
+            update_expression_parts=update_expression_parts['add'],
+            compact_action_additions=compact_action_additions,
+            jurisdiction_action_additions=jurisdiction_action_additions,
+        )
+        resp = (
+            self._handle_user_permission_removals(
+                user_id=user_id,
+                compact=compact,
+                update_expression_parts=update_expression_parts['delete'],
+                compact_action_removals=compact_action_removals,
+                jurisdiction_action_removals=jurisdiction_action_removals,
+            )
+            or resp
+        )
+
+        if not (update_expression_parts['add'] or update_expression_parts['delete']):
```

Comment:
why can't this be checked above, before the calls to `_handle_user_permission...`?

Reply:
The logic to compile the additions/removals collections into update expression parts is in their respective `_handle_user_permission...` methods, so the simplest check for whether no changes were requested is by looking at the two resulting expression parts, after those handlers are executed.